message.go

```diff
@@ -771,42 +747,30 @@ func (d *Decoder) Decode() (*Message, error) {
 	}
 
 	// Read segments.
-	if !d.reuse {
-		var buf []byte
-		if d.bufferPool == nil {
-			buf = make([]byte, int(total))
-		} else {
-			buf = d.bufferPool.Get(int(total))
-		}
-		if _, err := io.ReadFull(d.r, buf); err != nil {
-			return nil, exc.WrapError("decode: read segments", err)
-		}
-		arena, err := demuxArena(hdr, buf)
-		if err != nil {
-			return nil, exc.WrapError("decode", err)
-		}
-		return &Message{
-			Arena:          arena,
-			originalBuffer: buf,
-		}, nil
+	if d.reuse {
+		d.arena.Release()
+		d.arena = singleSegmentPool.Get().(Arena)
```

Comment:
Any reason not to just call `SingleSegment(nil)`?

Reply:
Fixed.